app/services/redirection_creation.rb

```diff
@@ -0,0 +1,25 @@
+class RedirectionCreation
+  def initialize(referrer, slug)
+    @referrer = referrer
+    @slug = slug
+  end
+
+  def perform
```

Comment:
I'd also recommend a class method to go along side this:

```ruby
def self.perform(referrer, slug)
  new(referrer, slug).perform
end
```

because it makes future testing (stubbing in particular) much easier. I.e. you don't have to stub `new` to return a double then stub out `perform` on the double. Also, if for whatever reason we use background jobs, I hear having those `perform` methods can make that easier too.